### PR TITLE
Small performance improvements

### PR DESCRIPTION
**Pull request by mrks**
... couple of things I found in the profiler:

Sort: Reserve vector
TPC-H generator: reserve space
Aggregate: reserve space before writing vector
ChunkOffsetMapping: Replace map by vector
replace murmur hash with std::hash after discussion with @Bouncner and finding that it does not improve the performance or the hash quality. Less code is always a good thing.

```
 Benchmark | prev. iter/s | new iter/s | change
-----------+--------------+------------+---------------
 TPC-H 1   | 3.808        | 3.821      | +0%
 TPC-H 2   | 89.732       | 92.27      | +3%
 TPC-H 3   | 39.409       | 40.101     | +2%
 TPC-H 4   | 16.387       | 16.897     | +3%
 TPC-H 5   | 26.012       | 27.707     | +7%
 TPC-H 6   | 60.353       | 81.409     | +35%
 TPC-H 7   | 7.564        | 9.184      | +21%
 TPC-H 8   | 33.835       | 36.258     | +7%
 TPC-H 9   | 16.567       | 16.24      | -2%
 TPC-H 10  | 21.306       | 21.832     | +2%
 TPC-H 11  | 118.809      | 123.5      | +4%
 TPC-H 12  | 35.046       | 40.904     | +17%
 TPC-H 13  | 24.746       | 24.445     | -1%
 TPC-H 14  | 86.308       | 111.459    | +29%
 TPC-H 15  | 40.739       | 59.588     | +46%
 TPC-H 16  | 17.256       | 17.511     | +1%
 TPC-H 17  | 4.384        | 4.322      | -1%
 TPC-H 18  | 3.351        | 3.842      | +15%
 TPC-H 19  | 2.426        | 2.463      | +2%
 TPC-H 20  | 0.031        | 0.031      | +0%
 TPC-H 21  | 0.293        | 0.39       | +33%
 TPC-H 22  | 49.897       | 50.574     | +1%
 average   |              |            | +10%

```

### CODE REVIEW

File: src/benchmarklib/tpch/tpch_db_generator.cpp [hunk 1]
```diff
@@ -191,21 +196,31 @@ TpchDbGenerator::TpchDbGenerator(float scale_factor, uint32_t chunk_size)
...
+  const auto part_count = static_cast<size_t>(tdefs[PART].base * _scale_factor);
+  const auto supplier_count = static_cast<size_t>(tdefs[SUPP].base * _scale_factor);
+  const auto nation_count = static_cast<size_t>(tdefs[NATION].base);
+  const auto region_count = static_cast<size_t>(tdefs[REGION].base);
+
+  TableBuilder customer_builder{_chunk_size, customer_column_types, customer_column_names, UseMvcc::Yes,
+                                customer_count};
+  TableBuilder order_builder{_chunk_size, order_column_types, order_column_names, UseMvcc::Yes, order_count};
+  TableBuilder lineitem_builder{_chunk_size, lineitem_column_types, lineitem_column_names, UseMvcc::Yes,
+                                order_count * 4};
```
Comment (timzimmermann) [inline src/benchmarklib/tpch/tpch_db_generator.cpp]: A comment/constant for the 4 might be helpful in the future.

Comment (timzimmermann) [reply src/benchmarklib/tpch/tpch_db_generator.cpp]: Same few lines down, not sure it's the same constant.

Comment (mrks) [reply src/benchmarklib/tpch/tpch_db_generator.cpp]: Not sure how to call that constant. It is a number by which the row count is multiplied and is defined as such in the TPC-H. I added a comment explaining that.

Comment (timzimmermann) [reply src/benchmarklib/tpch/tpch_db_generator.cpp]: I think that's good enough 👍

**Pull request merged**